src/python/pants/util/osutil.py

```diff
@@ -69,3 +69,11 @@ def known_os_names():
   ('darwin', '16'): ('mac', '10.12'),
   ('darwin', '17'): ('mac', '10.13'),
 }
+
+
+def get_closest_mac_host_platform_pair(darwin_version_upper_bound,
+                                       platform_name_map=SUPPORTED_PLATFORM_NORMALIZED_NAMES):
+  """Return the (host, platform) pair for the highest known darwin version less than the bound."""
+  darwin_versions = [int(x[1]) for x in platform_name_map if x[0] == 'darwin']
+  max_darwin_version = str(max(v for v in darwin_versions if v <= int(darwin_version_upper_bound)))
+  return platform_name_map[('darwin', max_darwin_version)]
```

Comment:
Using `get` and raising a more helpful error here would be good (or having the caller handle None and fall through to the other helpful error you added).

Reply:
Well, assuming we have any darwin versions at all lower than the upper bound then this can never error, so i'm not sure what a helpful message could even say?  And if we don't then max() will already have failed. I can check that this set isn't empty though.

Reply:
OK so this now falls through as you suggested.